### PR TITLE
Fix specs

### PR DESCRIPTION
**Pull request by nvandoorn**
These specs have not been run for a while. We want to fix a bug and add some new specs, so we should start by cleaning this up.

### CODE REVIEW

**Review by seriousammy — APPROVED** [review]
RIP factory girl.

I assume the specs being changed to pending are irrelevant to the current use of this repo? (I think I found that to be similar in other pervino repos, where certain flows were no longer being used.)

**Comment by nvandoorn** [comment]
Yeah they seem to be completely out of date with what the code is actually doing. I think we want to delete them or plan some time in a future cycle to fix them.

File: Gemfile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 source 'https://rubygems.org'
 
+gem "rails", "~> 5.2"
```
Comment (Noah-Silvera) [inline Gemfile:3]: As a gem, should we be defaulting to `RAILS_ENV` and then falling back to 5.2?

**Review by Noah-Silvera — APPROVED** [review]

File: Gemfile [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
```
Comment (alexblackie) [inline Gemfile:3]: There's no `Gemfile.lock` change for this, so it's still locked at 6.1.x and won't run

File: spec/models/spree/design_option_spec.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-RSpec.describe Spree::DesignOption, type: :model do
+RSpec.xdescribe Spree::DesignOption, type: :model do
```
Comment (alexblackie) [inline spec/models/spree/design_option_spec.rb:1]: IIRC some of these failures were just simple string suffix mismatches, would probably be quite easy to fix

**Review by alexblackie — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 2 above)
Comment (alexblackie) [reply Gemfile:3]: Apparently it's gitignored

File: Gemfile (hunk 1 above)
Comment (jarednorman) [reply Gemfile]: I think you probably mean `RAILS_VERSION`? I think that makes sense though, even if you only currently support "~> 5.2", to make developing against other versions for upgrades easier.

**Comment by nvandoorn** [comment]
Since last push:

- Support an environment variable for Rails version
- Took a stab at fixing most of the specs. They are passing but the rationale is a little complex so I'm saving it for another PR and leaving them as pending for now.

**Pull request merged**